Baikal/RenderFactory/clw_render_factory.h

```diff
@@ -55,7 +55,7 @@ namespace Baikal
             CreateOutput(std::uint32_t w, std::uint32_t h) const override;
         // Create post effect of specified type
         std::unique_ptr<PostEffect> 
-            CreatePostEffect(PostEffectType type) const override;
+            CreatePostEffect(PostEffectType type, std::size_t width, std::size_t height) const override;
```

Comment:
I common, I agree with you, but in the our case we can simplify client logic. Let's see to the old client code:
```
WaveletDenoiser* wavelet_denoiser = dynamic_cast<WaveletDenoiser*(m_outputs[m_primary].denoiser.get());
if (wavelet_denoiser != nullptr)
{
      wavelet_denoiser->Update(static_cast<PerspectiveCamera*>(m_camera.get()));
}

auto radius = 10U - RadeonRays::clamp((sample_cnt / 16), 1U, 9U);
auto position_sensitivity = 5.f + 10.f * (radius / 10.f);

const bool is_bilateral_denoiser = dynamic_cast<BilateralDenoiser*>(m_outputs[m_primary].denoiser.get()) != nullptr;

if (is_bilateral_denoiser)
{
      auto normal_sensitivity = 0.1f + (radius / 10.f) * 0.15f;
      auto color_sensitivity = (radius / 10.f) * 2.f;
      auto albedo_sensitivity = 0.5f + (radius / 10.f) * 0.5f;
      m_outputs[m_primary].denoiser->SetParameter("radius", static_cast<float>(radius));
      m_outputs[m_primary].denoiser->SetParameter("color_sensitivity", color_sensitivity);
      m_outputs[m_primary].denoiser->SetParameter("normal_sensitivity", normal_sensitivity);
      m_outputs[m_primary].denoiser->SetParameter("position_sensitivity", position_sensitivity);
      m_outputs[m_primary].denoiser->SetParameter("albedo_sensitivity", albedo_sensitivity);
}
```
As I can see, **BilateralDenoiser** depends only on _sample_cnt_ parameter, because _radius_, _color_sensitivity_, etc are some functions depending on _sample_cnt_. Therefore, such parameters are not interface parameters and they can be calculated inside BilateralDenoiser. So, we can provide generic interface method `void Update(Camera* camera, unsigned int samples)` to pass all need inforation to any kind od denoiser, and replace the code above to simple call:
`m_post_effect->Update(m_camera.get(), samples);`

Reply:
Is it valid case that PostEffect's width and height are differ from Output's width and height ?